### DISCUSSION TITLE
SqlSensor doesn't support BiqQuery using google_cloud_platform connection

### DISCUSSION THREAD

**Discussion by nenkie76** [apache/airflow#36845] [General] [answered]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

2.7.3

### What happened?

`SqlSensor` from `airflow.providers.common.sql.sensors.sql` doesn't support `BigQuery` when connection type is not `gcpbigquery`, resulting:
```
airflow.exceptions.AirflowException: The connection type is not supported by SqlSensor. The associated hook should be a subclass of `DbApiHook`. Got GoogleBaseHook
```

Although `gcpbigquery` connection doesn't have an option to keep JSON private key with service account.

### What you think should happen instead?

`SqlSensor` determines type of hook [here](https://github.com/apache/airflow/blob/f1243537838516b8bb8156130bc001595bfbeb01/airflow/providers/common/sql/sensors/sql.py#L85) using connection type and checking `DbApiHook` inheritance.

I can't use `SqlSensor` with `google_cloud_platform` connection defined, having a JSON private key and at the same time `gcpbigquery` connection type doesn't allow to keep private key. Though `BigQueryHook` itself [support](https://github.com/apache/airflow/blob/13b0930bf45faece076ef1b0c4e6fd14f2b17e16/airflow/providers/google/cloud/hooks/bigquery.py#L104C9-L104C20) `google_cloud_platform` connection. 

I have tried bypassing connection via `hook_params` like this:
```
SqlSensor(
   task_id="bq_sensor",
   conn_id="fake gcpbigquery connection",
   hook_params={
      "gcp_conn_id": "real google_cloud_platform  connection with PK",
      "location": "US"
   },
)
```
but it prevented [here](https://github.com/apache/airflow/blob/f1243537838516b8bb8156130bc001595bfbeb01/airflow/models/connection.py#L374) with message: 
```
TypeError: airflow.providers.google.cloud.hooks.bigquery.BigQueryHook() got multiple values for keyword argument 'gcp_conn_id'
```

Is there any other option to use `SqlSensor` with google service account?

### How to reproduce

See about explanation.

### Operating System

Mac

### Versions of Apache Airflow Providers

apache-airflow-providers-google 10.6.0

### Deployment

Astronomer

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
Converting it into discussion - it's not really an issue, it's just a question whether/how you can do stuff.

No - you must use `gcpbigquery` connection type because this is the connection type associated with BQHook and this is how SqlSensor knows which Hook to instantiate.

```
class BigQueryHook(GoogleBaseHook, DbApiHook):
    """Interact with BigQuery.

    This hook uses the Google Cloud connection.

    :param gcp_conn_id: The Airflow connection used for GCP credentials.
    :param use_legacy_sql: This specifies whether to use legacy SQL dialect.
    :param location: The location of the BigQuery resource.
    :param priority: Specifies a priority for the query.
        Possible values include INTERACTIVE and BATCH.
        The default value is INTERACTIVE.
    :param api_resource_configs: This contains params configuration applied for
        Google BigQuery jobs.
    :param impersonation_chain: This is the optional service account to
        impersonate using short term credentials.
    :param labels: The BigQuery resource label.
    """

    conn_name_attr = "gcp_conn_id"
    default_conn_name = "google_cloud_bigquery_default"
    conn_type = "gcpbigquery"
    hook_name = "Google Bigquery"
```

I am not sure where you get that the gcpbigquery connection does not support the private key though. It would be interesting to see how one support it where the other does not.

**Reply by nenkie76** [reply, marked as answer]
Hi @potiuk,

Well, just looking at `Admin -> Connections -> Add` and pick `BigQuery` connection type gives neither `Keyfile Path` nor `Keyfile JSON` field.
At the same time `Google Cloud` connection type has this option.

I tried creating `BigQuery` connection specifying `extras` like this and it works.

```
{
    "keyfile_dict": {
        "type": "service_account",
        "project_id": "...",
        ...
        "universe_domain": "googleapis.com"
    }
}
```

**Reply by potiuk** [reply]
So maybe it's a good reason to add it.?
